pandas/core/categorical.py

```diff
@@ -1386,17 +1386,16 @@ def unique(self):
         """
         Return the unique values.
 
-        Unused categories are NOT returned.
+        Unused categories are NOT returned. Unique values are returned in order
+        of appearance.
 
         Returns
         -------
         unique values : array
         """
-        unique_codes = np.unique(self.codes)
-        # for compatibility with normal unique, which has nan last
-        if unique_codes[0] == -1:
-            unique_codes[0:-1] = unique_codes[1:]
-            unique_codes[-1] = -1
+        from pandas.core.nanops import unique1d
```

Comment:
@jreback This is fed into `take_1d` which fills -1 with `fill_value`... which is happily exactly what we want here to handle NaN. That behavior is unchanged from before (and still tested). So I think this is OK?